script/import_scripts/smf2.rb

```diff
@@ -266,11 +269,121 @@ def guardian
           )
         end
       end
-      post[:raw] = convert_message_body(message[:body], attachments, ignore_quotes: ignore_quotes)
+      begin
+        post[:raw] = convert_message_body(message[:body], attachments, ignore_quotes: ignore_quotes)
+      rescue StandardError
+        post[:raw] = "-- MESSAGE SKIPPED --"
+      end
```

Comment:
What types of errors is this hiding? I fear that this error handling could lead to unexpected problems. Say, you import countless PMs, everything looks good, your site goes online, and only then do you notice that PMs weren't imported successfully.

I'd prefer to output useful information to the console. And ideally, we should try to handle problematic cases in `convert_message_body` so that it doesn't raise errors.

Reply:
The errors I received are probably due to some data being inherited from very old MySQL versions - as a matter of fact, only a couple of posts out of several hundred thousands caused errors; I am absolutely not competent enough with RoR (and Ruby in general) to be able to come up with a "good" way of handling them so I decided to just skip the posts in question.

Of course, I can remove these lines and let the errors simply break the process but, when it happened to me, that resulted in quite some time of addtional processing (the importer takes a long time to deal with existing posts).
Perhaps there's a way of making this conditional on a user's choice? A commandline parameter maybe?
If that's the case, I would appreciate very much a code suggestion, since - again - my skills here are extremely limited.

Reply:
Simplest solution:

```ruby
rescue => e
  puts "Failed to import message with ID #{post[:id]}"
  puts e.message
  puts e.backtrace.join("\n")
  post[:raw] = "-- MESSAGE SKIPPED --"
end
```